server/neptune/workflows/activities/execute.go

```diff
@@ -41,3 +48,50 @@ func (t *executeCommandActivities) ExecuteCommand(ctx context.Context, request E
 		Output: string(out),
 	}, nil
 }
+
+type StringCommand struct {
+	Command        string
+	Dir            string
+	AdditionalEnvs map[string]string
+}
+
+func (l *StringCommand) Load() (string, error) {
+	cmd := exec.Command("sh", "-c", l.Command) // #nosec
+	cmd.Dir = l.Dir
+
+	finalEnvVars := os.Environ()
+	for key, val := range l.AdditionalEnvs {
+		finalEnvVars = append(finalEnvVars, fmt.Sprintf("%s=%s", key, val))
+	}
+
+	cmd.Env = finalEnvVars
+	out, err := cmd.CombinedOutput()
+
+	if err != nil {
+		return "", errors.Wrap(err, "executing cmd")
+	}
+
+	// Trim newline from res to support running `echo env_value` which has
+	// a newline. We don't recommend users run echo -n env_value to remove the
+	// newline because -n doesn't work in the sh shell which is what we use
+	// to run commands.
+	// n
+	return strings.TrimSuffix(string(out), "\n"), err
+}
+
+// EnvVar is a container used for activity requests
+// and we use basic json serialization so we need to be explicit
```

Comment:
Not entire sure what this means, but should you have the json annotations on the fields then?

Reply:
Nah you don't need annotations unless your json field names differ from the struct field names